server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregationBuilder.java

```diff
@@ -123,6 +126,7 @@ public FiltersAggregationBuilder(StreamInput in) throws IOException {
         }
         otherBucket = in.readBoolean();
         otherBucketKey = in.readString();
+        keyedBucketInArray = in.readBoolean();
```

Comment:
This'll probably cause something like:
```
./gradlew -p qa/mixed-cluster/ -Dtests.method='*filter*'
```

to fail. The fix it to change it to something like:
```
keyedBucketInArray = in.getVersion().before(Version.V_8_5_0) ? false : in.readBoolean();
```

And to skip the write on the other side if the version is too old.